python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9063,6 +9063,152 @@ def test_dataframe_add_suffix():
+        {
+            "a": [1, 1, 2, 2],
+            "b": [1, 1, 2, 3],
+            "c": [2, 3, 4, 5],
+            "d": [6, 8, 9, 1],
+        }
+    ).set_index(["a", "b"])
+
+    actual = gdf.groupby(level=["a", "b"]).cov(min_periods=0, ddof=1)
+    expected = (
+        gdf.to_pandas().groupby(level=["a", "b"]).cov(min_periods=0, ddof=1)
+    )
+
+    assert_eq(expected, actual)
+
+
+def test_groupby_covariance_empty_columns():
+    gdf = cudf.DataFrame(columns=["id", "val1", "val2"])
+    pdf = gdf.to_pandas()
+
+    actual = gdf.groupby("id").cov(min_periods=0, ddof=1)
+    expected = pdf.groupby("id").cov(min_periods=0, ddof=1)
+
+    assert_eq(
+        expected, actual, check_dtype=False, check_index_type=False,
+    )
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "id": ["a", "a", "a", "b", "b", "b", "c", "c", "c"],
+            "val1": ["v", "n", "k", "l", "m", "i", "y", "r", "w"],
+            "val2": ["d", "d", "d", "e", "e", "e", "f", "f", "f"],
+        },
+        {
+            "id": ["a", "a", "a", "b", "b", "b", "c", "c", "c"],
+            "val1": [1, 1, 1, 2, 2, 2, 3, 3, 3],
+            "val2": ["d", "d", "d", "e", "e", "e", "f", "f", "f"],
+        },
+    ],
+)
+@pytest.mark.parametrize("gkey", ["id", "val1", "val2"])
+def test_groupby_cov_invalid_column_types(data, gkey):
+    with pytest.raises(
+        TypeError, match="Covariance accepts only numerical column-pairs",
+    ):
+        cudf.DataFrame(data).groupby(gkey).cov(min_periods=0, ddof=1)
+
+
+def test_groupby_cov_positive_semidefinite_matrix():
+    # Refer to discussions in PR #9889 re "pair-wise deletion" strategy
+    # being used in pandas to compute the covariance of a dataframe with
+    # rows containing
+    # Note: cuDF currently matches pandas behavior in that the covariance
+    # matrices are not guaranteed PSD (positive semi definite).
+    # https://github.com/rapidsai/cudf/pull/9889#discussion_r794158358
+    gdf = cudf.DataFrame(
+        [[1, 2], [None, 4], [5, None], [7, 8]], columns=["v0", "v1"]
+    )
+    actual = gdf.groupby(by=cudf.Series([1, 1, 1, 1])).cov()
+    actual.reset_index(drop=True, inplace=True)
+
+    pdf = gdf.to_pandas()
+    expected = pdf.groupby(by=pd.Series([1, 1, 1, 1])).cov()
+    expected.reset_index(drop=True, inplace=True)
+
+    assert_eq(
+        expected, actual, check_dtype=False,
+    )
+
+
+@pytest.mark.xfail
+def test_groupby_cov_for_pandas_bug_case():
+    # Handles case: pandas bug when min_periods=2 and ddof=2
+    # Currently pandas recognizes the bug with no clear fix
+    # Filed an issue in Pandas on GH, link below:
+    # https://github.com/pandas-dev/pandas/issues/45814
+    pdf = pd.DataFrame(
+        {
+            "id": ["a", "a", "b", "b", "c", "c"],
+            "val1": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0],
+            "val2": [np.nan, np.nan, np.nan, np.nan, np.nan, np.nan],
+        }
+    )
+    expected = pdf.groupby("id").cov(min_periods=2, ddof=2)
```

Comment:
`min_periods` does not matter for this bug.
```suggestion
    expected = pdf.groupby("id").cov(ddof=2)
```